### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by willespencer**
### Summary 

Implement onboarding to allow users to select their college, major, and minor before using the platform. 

Onboarding

- [x] Implemented the onboarding modal
- [x] Implement dropdowns based on the new semester modal, and allow both to be closed by clicking outside of them
- [x] Pull in the user's first and last name from their google account
- [x] Require certain fields to be filled out before submitting
- [x] Allow multiple majors and minors to be added or removed
- [x] Pull list of majors and colleges from reqs.json
- [x] Pass user data to the requirements menu so it displays based on Onboarding
- [x] Allow users to edit profile from the profile icon in the navbar

- [ ] Have a list of minors based on reqs.json
- [ ] Allow a user to select multiple colleges
- [ ] Pass multiple majors to requirements (data stored right now assumes 1 major and 1 college)

### Test Plan 

Select different combinations and numbers of colleges and majors. Ensure all data is saved after onboarding, including changing the name selected from the user's email. Ensure semesters and other changed information is saved after editing a profile.

### Other Notes 

Designs were not completely finished for a number of things, like adding/removing majors, editing profiles, etc. Because of this, some of the styling will likely need to change.

### Breaking Changes  

- Profiles on Firebase likely need to be deleted before logging into them with this change.

### CODE REVIEW

File: src/requirements/reqs.json [hunk 1]
```diff
@@ -544,514 +560,6 @@
...
-                                    "AEM 4580",
-                                    "AEM 4421"
-                                ]
-                            ],
-                            "fulfilledBy": "credits",
-                            "minCount": 9,
-                            "majorDep": true
-                        }
-                    ]
-                }
```
Comment (SamChou19815) [inline src/requirements/reqs.json]: What's happening here? Why so many lines are deleted?

File: src/components/Modals/Onboarding.vue [hunk 2]
```diff
@@ -0,0 +1,742 @@
+<template>
+  <div class="onboarding">
+    <div class="onboarding-main">
+      <div class="onboarding-content">
+        <div class="onboarding-top">
+          <div v-if="!isEditingProfile" class="onboarding-header">👏 Welcome to CoursePlan</div>
+          <div v-if="isEditingProfile" class="onboarding-header">👋 Hi {{ user.firstName }}</div>
```
Comment (SamChou19815) [inline src/components/Modals/Onboarding.vue]: Be careful about using emojis in UI. They look different in different OS!

File: src/components/Modals/Onboarding.vue [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+            </div>
+            <div class="onboarding-inputWrapper onboarding-inputWrapper--name">
+              <label class="onboarding-label">Middle Name</label>
+              <input class="onboarding-input" v-model="middleName" />
+            </div>
+            <div class="onboarding-inputWrapper onboarding-inputWrapper--name">
+              <label class="onboarding-label">Last Name*</label>
+              <input class="onboarding-input" v-model="lastName" />
+            </div>
+          </div>
```
Comment (SamChou19815) [inline src/components/Modals/Onboarding.vue]: Doesn't firebase auth already give you these info?

File: src/components/Requirements.vue [hunk 4]
```diff
@@ -6,13 +6,12 @@
...
-        <p class="name col p-0">{{ req.name }}</p>
-        <div class="col-1 text-right p-0">
+        <p class="name col p-0">{{ req.name }} <span class="specific" v-if="req.specific">({{ req.specific }})</span></p>
+        <!-- <div class="col-1 text-right p-0">
           <button @click="modalShow = !modalShow" class="btn">
-            <!-- svg for settings icon -->
             <img class="settings" src="../assets/images/gear.svg" />
           </button>
-        </div>
+        </div> -->
```
[the remark this answers is not in the document]
Comment (hanywang2) [reply src/components/Requirements.vue]: Yessir

File: src/requirements/reqs.json (hunk 1 above)
Comment (hanywang2) [reply src/requirements/reqs.json]: During the Planathon, Kathleen added a bunch of useful data for AEM concentration requirements. However, the current platform doesn't support concentrations within majors. We removed the data temporarily until our code is mature enough to support it. Have it saved and gonna get back to this soon enough

File: src/components/Modals/Onboarding.vue (hunk 2 above)
Comment (hanywang2) [reply src/components/Modals/Onboarding.vue]: Thanks for the heads up

File: src/components/Modals/Onboarding.vue (hunk 3 above)
Comment (hanywang2) [reply src/components/Modals/Onboarding.vue]: It's a discussion the design team has had before (and also again with me and Will recently). Letting them enter their own data enhances the 'personalized' feel of the platform. Evan also pointed out that this accounts for international students, those with nicknames, and members of the LGBTQ+.

**Pull request merged**